test/test_torch.py

```diff
@@ -6987,80 +6988,74 @@ def test_logical_and(self, device):
     def test_logical_or(self, device):
         self._test_logical(device, 'logical_or', [10, 0, 1, 0], [1, 0, 0, 10], [1, 0, 1, 1])
 
-    def test_clamp(self, device):
-        m1 = torch.rand(100, device=device).mul(5).add(-2.5)  # uniform in [-2.5, 2.5]
-        # just in case we're extremely lucky.
-        min_val = -1
-        max_val = 1
-        m1[1] = min_val
-        m1[2] = max_val
+    def generate_clamp_baseline(self, device, min_vals=None, max_vals=None, with_nans=False):
+        X = torch.rand(100, device=device).mul(5).add(-2.5)  # uniform in [-2.5, 2.5]
+        if with_nans:
+            mask = torch.randint(0, 2, X.shape, dtype=torch.bool, device=device)
+            X[mask] = nan
 
-        res1 = m1.clone()
-        res1.clamp_(min_val, max_val)
-        res2 = m1.clone()
-        for i in iter_indices(res2):
-            res2[i] = max(min_val, min(max_val, res2[i]))
-        self.assertEqual(res1, res2)
+        if isinstance(min_vals, torch.Tensor):
+            min_vals = min_vals.cpu().numpy()
 
-        out = m1.clone()
-        torch.clamp(m1, min=min_val, max=max_val, out=out)
-        self.assertEqual(out, res1)
+        if isinstance(max_vals, torch.Tensor):
+            max_vals = max_vals.cpu().numpy()
 
-        res1 = torch.clamp(m1, min=min_val)
-        res2 = m1.clone()
-        for i in iter_indices(res2):
-            res2[i] = max(min_val, res2[i])
-        self.assertEqual(res1, res2)
+        # Use NumPy implementation as reference
+        X_clamped = torch.tensor(np.clip(X.cpu().numpy(), a_min=min_vals, a_max=max_vals))
+        return X, X_clamped
 
-        torch.clamp(m1, min=min_val, out=out)
-        self.assertEqual(out, res1)
+    def test_clamp(self, device):
+        # Create min/max argument product
+        args = product((torch.linspace(-1, 1, 100, device=device), -1, None),
```

Comment:
How about generating two randn tensors instead of using linspace? Otherwise when the two tensors are selected you're only testing the narrow case when min=max, right?

Reply:
Good spot! That was a copy-paste/refactoring issue, it was meant to be `linspace(-1, 1, 100)` and `linspace(1, -1, 100)`, such that all cases are tested, I will fix it. I don't believe the randn is necessary here.